### PR TITLE
added unit, integration and end to end test to backend

### PR DESCRIPTION
**Pull request by LordFarquaadtheCreator**
did this with rafid. github action test will not work - unfortunately did not have enough time to implement correctly. All other tests work perfectly locally given the database connects (only whitelisted certain IPs)

### CODE REVIEW

**Comment by LordFarquaadtheCreator** [comment]
Test Coverage (backend):
Unit tests - tests to see if dummy endpoint `create_exercise` properly checks the type of input received (both correct and incorrect inputs checked).

Integration test - tests to see if developed endpoint `get_exercise` properly checks if the inputs are correct - then successfully sends back result from database.

Test Execution Automation - modified github workflow folder now runs the tests in the backend - however there is an issue where `jest` does not execute the tests properly and causes all tests to fail. 

End-To-End Test - unfortunately this test was not implemented due to the lack of development of the front-end. We did manually test a button on the UI that accesses the `get_exercise` endpoint - but were not able to setup test software to accomplish this at this time.

File: .DS_Store
Comment (navteniev) [inline .DS_Store:1]: This is a binary file crated by mac os machines, it needs to be excluded form the repository.

File: MuscleMate/test_output.json
Comment (navteniev) [inline MuscleMate/test_output.json:1]: This is a generated file and must not be checked in.

File: backend/test_output.json
Comment (navteniev) [inline backend/test_output.json:1]: Same comment as above no need to check in automatically generated files.

File: backend/tests/server.test.ts [hunk 1]
```diff
@@ -11,22 +11,47 @@ describe('Server Actions', () => {
...
             await request(app).get('/').expect(200);
         });
     });
-    describe('Get Exercises', () => {
-        const keywords = ['value1', 'value2'];
-        const keywordsString = encodeURIComponent(JSON.stringify(keywords));
-        
+
+    describe('Post /create_exercise', () => {
+        // test('valid input', async () => {
```
Comment (navteniev) [inline backend/tests/server.test.ts:23]: what is the benefit of committing commented out code?

**Review by navteniev — COMMENTED** [review]
Unit test: 0/2 (no unit tests)
Integration test: 1/2 (only failure scenarios tested)
end-to-end test: 0/1 (no tests)
CI: 2/4 (not uploading rest result artifacts, not running end to end tests)
PR 1/1 (however you need to exclude auto generated files and binary files)

Final 4/10 pts.

**Pull request closed without merging**